File: src/main/java/no/jstien/bikeapi/station/api/BikeAPI.java
package no.jstien.bikeapi.station.api;

import no.jstien.bikeapi.station.Station;
import no.jstien.bikeapi.station.StationAvailability;

import java.util.List;

public interface BikeAPI {
    boolean areAllStationsClosed();
    List<Station> getAllStations();
    List<StationAvailability> getAvailabilities();
}

File: src/main/java/no/jstien/bikeapi/station/api/BikeAPI.kt
package no.jstien.bikeapi.station.api

import no.jstien.bikeapi.station.Station
import no.jstien.bikeapi.station.StationAvailability

interface BikeAPI {
    fun getAllStations(): List<Station>
    fun getAvailabilities(): List<StationAvailability>
    fun areAllStationsClosed(): Boolean
}
